lib/creditsafe/client.rb

```diff
@@ -127,15 +127,29 @@ def handle_message_for_response(response)
       end
     end
 
-    # Takes a proc and rescues any SOAP faults, HTTP errors or Creditsafe errors
     # There's a potential bug in the creditsafe API where they actually return
     # an HTTP 401 if you're unauthorized, hence the sad special case below
-    def wrap_soap_errors
-      response = yield
+    def invoke_soap(message_type, message)
+      started = Time.now
+      notification_payload = { request: message }
+
+      response = client.call(message_type, message: message)
       handle_message_for_response(response)
-      response.body
-    rescue => error
-      handle_error(error)
+      notification_payload[:response] = response.body
+    rescue => raw_error
+      begin
+        handle_error(raw_error)
+      rescue => processed_error
+        notification_payload[:error] = processed_error
+        raise processed_error
+      end
```

Comment:
the error ping-pong going on here is quite silly, but I guess it's a little better than case statement shenanigans in handle_error? maybe?